File: pilhas-filas/javascript/stack2-test.js
import Stack2 from "./stack2.js";

let stack = new Stack2();

console.log("stack.count(): " + stack.count())
console.log("stack.isEmpty(): " + stack.isEmpty());

stack.push("Maria");
stack.push("Joao");
stack.push("Ana");
console.log("Adicionados: Maria, Joao, Ana");

console.log("stack.count(): " + stack.count())
console.log("stack.isEmpty(): " + stack.isEmpty());
console.log("stack.peek(): " + stack.peek());

console.log("Itens:");
while (!stack.isEmpty()) {
  const item = stack.pop();
  console.log(item);
}

File: pilhas-filas/python/stack2_test.py
from stack2 import Stack2

stack = Stack2()

print(f"stack.count(): {stack.count()}")
print(f"stack.is_empty(): {stack.is_empty()}")

stack.push("Maria")
stack.push("Joao")
stack.push("Ana")
print("Adicionados: Maria, Joao, Ana")

print(f"stack.count(): {stack.count()}")
print(f"stack.is_empty(): {stack.is_empty()}")
print(f"stack.peek(): {stack.peek()}")

print("Itens:")
while not stack.is_empty():
    print(stack.pop())
